### PR TITLE
[ExpressionLanguage] [5.0] add type-hints whenever possible

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | 5.0
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | Contribute to #32179   
| License       | MIT
| Doc PR        | not needed 

Add type-hint whenever possible.

### CODE REVIEW

File: src/Symfony/Component/ExpressionLanguage/ExpressionFunction.php [hunk 1]
```diff
@@ -64,16 +64,13 @@ public function getEvaluator()
     /**
      * Creates an ExpressionFunction from a PHP function name.
      *
-     * @param string      $phpFunctionName        The PHP function name
-     * @param string|null $expressionFunctionName The expression function name (default: same than the PHP function name)
```
Comment (fabpot) [inline src/Symfony/Component/ExpressionLanguage/ExpressionFunction.php]: I would keep this one as the description is useful.

File: src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php [hunk 2]
```diff
@@ -104,15 +101,11 @@ public function parse($expression, $names)
     /**
      * Registers a function.
      *
-     * @param string   $name      The function name
-     * @param callable $compiler  A callable able to compile the function
-     * @param callable $evaluator A callable able to evaluate the function
```
Comment (fabpot) [inline src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php]: Same here, the 2 last descriptions are interesting IMHO

File: src/Symfony/Component/ExpressionLanguage/TokenStream.php [hunk 3]
```diff
@@ -58,11 +58,9 @@ public function next()
     /**
      * Tests a token.
      *
-     * @param array|int   $type    The type to test
-     * @param string|null $value   The token value
-     * @param string|null $message The syntax error message
```
Comment (fabpot) [inline src/Symfony/Component/ExpressionLanguage/TokenStream.php]: Same here

**Review by fabpot — CHANGES_REQUESTED** [review]

**Comment by Simperfit** [comment]
Status: Needs Review

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Simperfit.

**Pull request merged**